examples/plot_depth_classification.py

```diff
@@ -0,0 +1,331 @@
+"""
+Depth based classification
+==========================
+
+This example shows the use of the depth based classifications methods
+applied to the Berkeley Growth Study data. An attempt to show the
+differences and similarities between `MaximumDepthClassifier`,
+`DDClassifier`, and `DDGClassifier` is made.
+"""
+
+# Author: Pedro Martín Rodríguez-Ponga Eyriès
+# License: MIT
+
+# sphinx_gallery_thumbnail_number = 5
+
+import matplotlib.pyplot as plt
+import numpy as np
+from matplotlib.colors import ListedColormap
+from sklearn.model_selection import train_test_split
+from sklearn.neighbors import KNeighborsClassifier
+from sklearn.neural_network import MLPClassifier
+
+from skfda import datasets
+from skfda.exploratory.depth import ModifiedBandDepth
+from skfda.exploratory.visualization import DDPlot
+from skfda.ml.classification import (
+    DDClassifier,
+    DDGClassifier,
+    MaximumDepthClassifier,
+)
+from skfda.preprocessing.dim_reduction.feature_extraction import DDGTransformer
+from skfda.representation.grid import FDataGrid
+
+##############################################################################
+#
+# The Berkeley Growth Study data contains the heights of 39 boys and 54 girls
+# from age 1 to 18 and the ages at which they were collected. Males are
+# assigned the numeric value 0 while females are coded to a 1. In our
+# comparison of the different methods, we will try to learn the sex of a person
+# by using its growth curve.
+X, y = datasets.fetch_growth(return_X_y=True, as_frame=True)
+X = X.iloc[:, 0].values
+categories = y.values.categories
+y = y.values.codes
+
+##############################################################################
+#
+# As in many ML algorithms, we split the dataset into train and test. In this
+# graph, we can see the training dataset. These growth curves will be used to
+# train the model. Hence, the predictions will be data-driven.
+X_train, X_test, y_train, y_test = train_test_split(
+    X,
+    y,
+    test_size=0.5,
+    stratify=y,
+    random_state=0,
+)
+
+# Plot samples grouped by sex
+X_train.plot(group=y_train, group_names=categories).show()
+
+##############################################################################
+#
+# Below are the growth graphs of those individuals that we would like to
+# classify. Some of them will be male and some female.
+X_test.plot().show()
+
+##############################################################################
+#
+# As said above, we are trying to compare three different methods:
+# MaximumDepthClassifier, DDClassifier, and `DDGClassifier`. They all use a
```

Comment:
Use the `:class:` role to put links to the docs for the depth classification and visualization classes. There are several instances of these in this example.